src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -364,16 +366,13 @@ def _monitor_workers_and_process_tasks_in_thread(
         # If the while loop exits due to batch run timeout, we should set is_timeout to True if we didn't set it before.
         self._is_timeout = self._is_timeout or self._batch_timeout_expired(batch_start_time)
 
-        # wait 10 seconds to ensure the spans are exported before the process is killed
-        # otherwise there will be some missing spans for lines
-        # TODO (Task 2950080): make line process wait for spans flush
-        from promptflow._trace._start_trace import _is_tracer_provider_configured
-
-        if _is_tracer_provider_configured():
-            time.sleep(10)
+        if not self._is_timeout:
+            input_queue.put(TERMINATE_SIGNAL)
+            time.sleep(1)
```

Comment:
join?

Reply:
Maybe we don't need to sleep here, we only need to check the status in end_process

Reply:
Refined